src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -298,6 +371,34 @@ private static void ChangeColor(bool foreground, ConsoleColor color)
         /// <summary>Cache of the format strings for foreground/background and ConsoleColor.</summary>
         private static readonly string[,] s_fgbgAndColorStrings = new string[2, 16]; // 2 == fg vs bg, 16 == ConsoleColor values
 
+        public static bool TryGetSpecialConsoleKey(char[] givenChars, int startIndex, int endIndex, out ConsoleKey key, out int keyLength)
+        {
+            key = (ConsoleKey)0;
+            keyLength = 0;
+            int unprocessedCharCount = endIndex - startIndex + 1;
+
+            if (unprocessedCharCount < TerminalKeyInfo.Instance.MinKeyLength)
```

Comment:
> Why would we need to?

Our motivation for this change is accuracy. What if the last thing we captured was Esc and the next bit is J(when we poll), we will interpret this as enter with our new implementation and in our old one it would have been Esc and J. Any of this could have been true. My point is both implementations could give wrong results, since we are second guessing here. I am open to changing our implementation to what you suggest, I am just not sure if it is more accurate than the current one.